### PR TITLE
vimPlugins.vim-mediawiki-editor: init at 2022-10-29

### PR DESCRIPTION
**Pull request by tomodachi94**
###### Description of changes

Add `vimPlugins.vim-mediawiki-editor` to plugins list, update plugins list ~~and tree-sitter parsers~~.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by tomodachi94** [comment]
cc @figsoda, the tree-sitter parsers updated this time. Not sure what happened over in #208664.

**Comment by tomodachi94** [comment]
Ah, I forgot to add overrides for the dependencies. Give me a moment.

**Comment by figsoda** [comment]
The grammar update has an empty, you can drop it safely

I will fix the update script in another pr

**Comment by tomodachi94** [comment]
@figsoda, I've dropped the grammars commit and added dependencies. This PR is ready.

File: pkgs/applications/editors/vim/plugins/overrides.nix [hunk 1]
```diff
@@ -669,6 +670,10 @@ self: super: {
     dontBuild = true;
   });
 
+  vim-mediawiki-editor = super.vim-mediawiki-editor.overrideAttrs (old: {
+    dependencies = with self; [ python310Packages.mwclient ];
```
Comment (figsoda) [inline pkgs/applications/editors/vim/plugins/overrides.nix:674]: this is probably what you want
```suggestion
    passthru.python3Dependencies = [ python3.pkgs.mwclient ];
```

Comment (tomodachi94) [reply pkgs/applications/editors/vim/plugins/overrides.nix]: Yep, nice catch. Added in the latest push.

**Review by figsoda — APPROVED** [review]

**Pull request merged**